src/main/java/com/weirddev/testme/intellij/template/context/MockitoMockBuilder.java

```diff
@@ -108,12 +108,27 @@ private Integer safeParseInteger(String intStr) {
      * true - field can be mocked
      */
     @SuppressWarnings("unused")
-    public boolean isMockable(Field field) {
-        final boolean isMockable = !field.getType().isPrimitive() && !isWrapperType(field.getType()) && (!field.getType().isFinal() || isMockitoMockMakerInlineOn) && !field.isOverridden() && !field.getType().isArray() && !field.getType().isEnum();
+    public boolean isMockable(Field field, Type testedClass) {
+        final boolean isMockable = !field.getType().isPrimitive() && !isWrapperType(field.getType()) && (!field.getType().isFinal() || isMockitoMockMakerInlineOn) && !field.isOverridden() && !field.getType().isArray() && !field.getType().isEnum() && !isNotMockableViaDi(field, testedClass);
         LOG.debug("field "+field.getType().getCanonicalName()+" "+field.getName()+" is mockable:"+isMockable);
         return isMockable;
     }
 
+    /**
+     * Avoid mocking a field if all these conditions are met:
+     * class has DI annotation
+     * field has no direct DI annotation
+     * field does not have a setter
+     * there is no class constructor.
+     * @param field  field
+     * @param testedClass testedClass
+     * @return true if field meet conditions above
+     */
+    public boolean isNotMockableViaDi(Field field, Type testedClass) {
```

Comment:
got confused by the method name. can you please rename, perhaps to _isNotInjectedInDiClass_